### ISSUE TITLE
UI should mark and understand different files

### ISSUE THREAD

**Issue by knmnyn** [WING-NUS/SSID#8]
SSID concatenates all files to do the plagiarism detection.  However, information about which files are which are lost, and should be presented in the UI.

**Comment by knmnyn** [comment]
Marking boundaries of files.  
Add a `SSID_FILE_BOUNDARY`

**Comment by knmnyn** [comment]
Add three lines 
`FILE A --- SSID_END_FILE`
`SSID_FILE_BOUNDARY`
`FILE B --- SSID_BEGIN_FILE`

**Comment by YisongMiao** [comment]
Issue solved (not evaluated!) for a specific language (python): https://github.com/YisongMiao/SSID/commit/118e07665102f06e2e62421bdb91f5ad1bece557

**Comment by YisongMiao** [comment]
> Issue solved (not evaluated!) for a specific language (python): [YisongMiao@118e076](https://github.com/YisongMiao/SSID/commit/118e07665102f06e2e62421bdb91f5ad1bece557)

The key idea for the implementation:

**Comment by YisongMiao** [comment]
Min's comment: 1. it's a short-term solution, long-term should go for the data structure; 2. can add an agnostic function to convert lines into comment

**Comment by YisongMiao** [comment]
Generally it works fine! 
I evaluate my last week's solution on the lab PC, following are some tests:

![Screenshot from 2021-08-03 14-14-57](https://user-images.githubusercontent.com/11874006/127967000-9e512170-9db5-4425-80eb-57e6e81345d0.png)

From the figure above, we can see the SSID_BEGIN_FILE is created.

![Screenshot from 2021-08-03 14-17-32](https://user-images.githubusercontent.com/11874006/127967310-2fac8a56-75a9-43fe-a62c-fea501f96b40.png)

From the figure above, we can see the segmentation lines are also inserted.

Another key properties to study is whether these lines have biased the plagiarism detection. Our observation so far is that it is not biased.

![Screenshot from 2021-08-03 14-21-38](https://user-images.githubusercontent.com/11874006/127967729-53205d0d-deac-4157-b770-e445df8d8ea8.png)

As shown in the figure above, we can clear see that the segmentation lines are not highlighted.

In conclusion, so far so good. I then need to expand it to support other languages.

**Comment by YisongMiao** [comment]
Add file support other than Python. Need to evaluate ... https://github.com/YisongMiao/SSID/commit/53e5af44badd3f0f736e74830ccd560d4e8e04e3

**Comment by YisongMiao** [comment]
![Screenshot from 2021-08-17 17-55-17](https://user-images.githubusercontent.com/11874006/129705743-94b16a28-2391-4013-a92b-07ed35e44258.png)
![Screenshot from 2021-08-17 17-56-05](https://user-images.githubusercontent.com/11874006/129705753-dff95517-cab1-4a9d-96f7-c2470c7f2b67.png)
![Screenshot from 2021-08-17 17-57-21](https://user-images.githubusercontent.com/11874006/129705765-28fe899c-99cd-4298-a782-31a6d556ecc4.png)

So now we can see the segmentation lines are supporting `m` `java`, and `c`. They are all working fine and not biasing the plagiarism detection engine. So issue close!
